File: Code/search_algorithms/exponential_search/exponential_search.js
const binarySearch = (arr, l, r, x) => {
    if (r >= l) {
        mid = l + (r - l) / 2
 
        if (arr[mid] == x)
            return mid
 
        if (arr[mid] > x)
            return binarySearch(arr, l, mid-1, x)
 
        return binarySearch(arr, mid+1, r, x)
    }
    return -1
};

const exponentialSearch = (arr, x) => {
    n = arr.length
    if (arr[0] == x)
        return 0

    i = 1
    while (i < n && arr[i] <= x)
        i *= 2
 
    return binarySearch(arr, i/2, Math.min(i, n-1), x)
};

arr = [1, 3, 4, 10, 40]
x = 10
result = exponentialSearch(arr, x)

console.log("Element is", result === -1 ? "not present in array": `present at index ${result}`)

File: Code/search_algorithms/exponential_search/exponentialSearch.py
def binarySearch(ar, l, r, x):
    if r >= l:
        mid = int(l + (r - l) / 2)
        if ar[mid] == x:
            return mid
        if ar[mid] > x:
            return binarySearch(ar, l, mid - 1, x)
        return binarySearch(ar, mid + 1, r, x)
    return -1


def exponentialSearch(ar, n, x):
    if ar[0] == x:
        return 0
    i = 1
    while i < n and ar[i] <= x:
        i = i * 2
    return binarySearch(ar, i / 2, min(i, n - 1), x)


ar = [2, 3, 4, 10, 40]
n = len(ar)
x = 10
result = exponentialSearch(ar, n, x)
if result == -1:
    print("Element not the array")
else:
    print(str(x) + " is at index %d" % result)
